crates/flashblocks-rpc/src/rpc.rs

```diff
@@ -387,7 +395,41 @@ where
                         block_number: Some(block_number),
                         base_fee,
                     };
-                    Ok(Some(self.transform_tx(transaction, tx_info)))
+                    // preload transaction receipt if it's a deposit transaction
+                    if transaction.is_deposit() {
```

Comment:
Why did all this move to this method and pass in the receipt? Why not keep this logic in the `transform_tx` and prevent the addition of the receipt param? That way if the method is used anywhere else we won't have to replicate this logic. Otherwise looks good

Reply:
It will color all functions async that use it. This was done pretty much to avoid marking everything async